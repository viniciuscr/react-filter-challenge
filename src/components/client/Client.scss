@import "../../colors.scss";
h4 {
  color: $light-blue;
  font-weight: 600;
  & > small.text-muted {
    font-weight: 600;
  }
}
.badge {
  color: white;
  padding: 0.3rem;
  width: 6rem;
  text-transform: capitalize;
}

.active {
  background-color: $green;
}

.canceled {
  background-color: red;
}

.producsts {
  text-transform: capitalize;
  list-style-type: none;
  margin: 1rem 0 1rem 0;
  padding: 0;
  font-weight: 600;
}

.bt-foward {
  background-color: unset;
  border: unset;
  padding: 1rem;
  border-left: 1px solid $light-gray;
  color: $gray;
}

.client-box {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-content: center;
}
