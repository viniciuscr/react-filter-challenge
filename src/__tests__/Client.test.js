import React from "react";
import { mount } from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Client from "../components/client";

configure({ adapter: new Adapter() });

describe("Client tests", () => {
  const clientData = {
    name: "My Realty",
    id: "236244",
    products: [
      {
        product_handle: "lead_management",
        state: "active"
      },
      {
        product_handle: "dashboard",
        state: "canceled"
      }
    ]
  };

  it("should render as expected", () => {
    const wrapper = mount(<Client>{clientData}</Client>);
    expect(wrapper.html()).toMatchSnapshot();
    wrapper.unmount();
  });
});
