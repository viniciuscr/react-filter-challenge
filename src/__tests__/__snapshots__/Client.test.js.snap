// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Client tests should render as expected 1`] = `"<div class=\\"shadow-sm h-100  mb-2 border bg-white rounded-sm \\"><div class=\\"client-box\\"><div class=\\"container-fluid\\"><h4>My Realty <small class=\\"text-muted\\">236244</small></h4><ul class=\\"producsts\\"><li><span class=\\"badge active\\">active</span> lead_management</li><li><span class=\\"badge canceled\\">canceled</span> dashboard</li></ul></div><button class=\\"bt-foward\\"><span class=\\"icon-rl-forward\\"></span></button></div></div>"`;
